src/audio/stft_process/stft_process.h

```diff
@@ -0,0 +1,239 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2025 Intel Corporation.
+ *
+ */
+#ifndef __SOF_AUDIO_STFT_PROCESS_H__
+#define __SOF_AUDIO_STFT_PROCESS_H__
+
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/math/auditory.h>
+#include <sof/math/dct.h>
+#include <sof/math/fft.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#if CONFIG_LIBRARY
+#define STFT_DEBUG	0	/* Keep zero, produces large files with fprintf() */
+#else
+#define STFT_DEBUG	0
+#endif
+
+#define SOF_STFT_PROCESS_CONFIG_MAX_SIZE 256 /* Max size for configuration data in bytes */
+
+enum sof_stft_process_fft_pad_type {
+	STFT_PAD_END = 0,
+	STFT_PAD_CENTER = 1,
+	STFT_PAD_START = 2,
+};
+
+enum sof_stft_process_fft_window_type {
```

Comment:
Assume the windows type and size (configuration) can be wholly specified by topology if needed ?